src/builtin/cd.c

```diff
@@ -6,25 +6,24 @@
 /*   By: msloot <msloot@student.42.fr>              +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/10/29 17:42:45 by msloot            #+#    #+#             */
-/*   Updated: 2024/11/01 17:17:48 by msloot           ###   ########.fr       */
+/*   Updated: 2024/12/02 22:15:31 by msloot           ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "minishell.h"
 
-int	cd(size_t argc, char **argv, char **envp)
+int	cd(t_args *arg)
 {
 	int	ret;
 
 	ret = 0;
-	if (argc > 2)
-		return (error(argv[0], "too many arguments"), EX_ERR);
-	if (argc == 1)
+	if (arg->argc > 2)
+		return (error(arg->argv[0], "too many arguments"), D_ERROR);
```

Comment:
```suggestion
		return (error(arg->argv[0], "too many arguments"), EX_ERR);
```

`D_XXX` is only for the `dispatch`

all `builtin` are like binary, they return an `exit code` that is `EX_XXX`